Fix pygobject CI for windows

soreau opened the pull request:
Since it's a small world, I decided to try my hand at fixing your CI. [Here](https://github.com/soreau/bbgtkwin/actions/runs/13298352815/artifacts/2582737474) is the resulting wheel link. Hopefully everything works out. Cheers. 👍

soreau commented:
> The GitHub image already has MSVC++. **Do you know to get the wheel built with MSVC++ 2022**?

As far as I could tell, this is not possible due to the fact that glib defines `g_autofree` only if [this condition](https://gitlab.gnome.org/GNOME/glib/-/blob/main/glib/gmacros.h#L1339) is true, which is not the case with `cl` (see [here](https://gitlab.gnome.org/GNOME/glib/-/blob/main/glib/gmacros.h#L1395)). This means that trying to build with a compiler that does not support `__attribute__((__cleanup__))` will not work. I'm guessing at this point, pygobject only works with gcc (and probably clang), but not cl, without further patching.

soreau commented:
So I patched pygobject, and now it built with `cl`. Does [this wheel](https://github.com/soreau/bbgtkwin/actions/runs/13300829159/artifacts/2583418381) work for you?

az0 commented:
> I'm guessing at this point, pygobject only works with gcc (and probably clang), but not cl, without further patching.

tl;dr I'm already using an older version of pyobject made by MSVC.

BleachBit 4.9.0 alpha uses using Python 3.10 with [PyGObject-3.42.0-cp310-cp310-win32.whl](  https://github.com/mkhon/vcpkg/releases/download/gtk3-introspection-v1/PyGObject-3.42.0-cp310-cp310-win32.whl) as used in this [Powershell script](https://github.com/bleachbit/bleachbit/blob/6b13a73c4722b5c50499697348ed5350b995cd8e/windows/python-gtk3-install.ps1) that sets up the Python+GTK environment. When I unzip this .whl file and run `objdump -p gi/_gi.cp310-win32.pyd ` , there are signs it uses MSVC++
* Linker Version 14.28
* VCRUNTIME140.dll dependency

Max provided these Python, GTK, and PyGObject builds, and this repository attempts to update and rebuild them from source. I know he used vcpkg and upstreamed patches to vcpkg, but otherwise, I am not sure about the details. He's not available at the moment to explain.

>So I patched pygobject, and now it built with cl. Does [this wheel](https://github.com/soreau/bbgtkwin/actions/runs/13300829159/artifacts/2583418381) work for you?

I will check it out. Thanks again!

soreau commented:
> > So I patched pygobject, and now it built with cl. Does [this wheel](https://github.com/soreau/bbgtkwin/actions/runs/13300829159/artifacts/2583418381) work for you?
> 
> I will check it out. Thanks again!

To be clear, [this](https://termbin.com/8saf) is the patch used and [this](https://github.com/soreau/bbgtkwin/actions/runs/13300829159/job/37141848522) is the CI run. If the resulting wheel works, I can update this PR.

az0 commented:
Yes, the wheel is working! This will fix a long-standing issue that BleachBit had old versions of Python and GTK, as they were hard to build and upgrade. This is finally reproducible.

>If the resulting wheel works, I can update this PR. 

I'll add two code comments and wait to hear back

soreau commented:
> Yes, the wheel is working! This will fix a long-standing issue that BleachBit had old versions of Python and GTK, as they were hard to build and upgrade. This is finally reproducible.

That's great!

Since the target is a pygobject release, the patch should apply cleanly regardless what happens upstream. So it should be good until the next pygobject upgrade at least, at which point it may have been patched to support `cl`. The question I think, is where should the patch be stored? The current patch link is a temporary one, so I propose that you take ownership of it in the form of a gist. If you can upload it to a github gist and share the link here, I can finish up this pull request 'properly'.

az0 commented:
If your GI patch can be in this (bbgtkwin) repository, I would do that because it's self-contained. Otherwise, here is the gist https://gist.github.com/az0/d252f29d68e705d6b65da123f1678fa7

The three lines I added about python311.lib may not be needed. If you prefer, I'll remove them after merging.

I'd like to retain the "release when tag is pushed" functionality, but again, I can fix that after merging.

BTW, I plan to add a testing step to install the wheel run "import gi" in python.exe

soreau commented:
> If your GI patch can be in this (bbgtkwin) repository, I would do that because it's self-contained. Otherwise, here is the gist https://gist.github.com/az0/d252f29d68e705d6b65da123f1678fa7

I think that's actually a great idea. The only trouble with this is, the patch would need to be part of the repository so I can link to it in the CI. Otherwise, if I commit it in this PR, the url will have to be adjusted. So, if you want to go ahead and commit the patch (as a .diff maybe), then I could link to it in the CI in this pull request.

> The three lines I added about python311.lib may not be needed. If you prefer, I'll remove them after merging.

I think they are required, at least as far as I can recall from my tests.

> I'd like to retain the "release when tag is pushed" functionality, but again, I can fix that after merging.

Yes, I removed this but can add it back.

> BTW, I plan to add a testing step to install the wheel run "import gi" in python.exe

Sounds good to me. 👍

I'll work on this and push to my branch to update this pull request, when ready.

mkhon commented:
1) I am not sure pkgconfig from Choco is required – it is comes with vcpkg

2) I do not remember I ever needed any pygtk patch to build it before

soreau commented:
> 1. I am not sure pkgconfig from Choco is required – it is comes with vcpkg
> 
> 2. I do not remember I ever needed any pygtk patch to build it before

Please submit a CI patch if these items are possible.

Reference for the reasoning behind the **pygobject** patch is in a [comment above](https://github.com/bleachbit/pygtkwin/pull/1#issuecomment-2655428376).

az0 commented:
> 1. I am not sure pkgconfig from Choco is required – it is comes with vcpkg

I added [pkgconfig to vcpkg.json](https://github.com/bleachbit/pygtkwin/commit/8a3bad5cab33d8181a3e6b0771918c1af82d440a), but I couldn't get meson to build pygobject. It may have failed for other reasons, though.

soreau commented:
It looks like it found [strawberry pkg-config](https://github.com/bleachbit/pygtkwin/actions/runs/13212691037/job/36909564893#step:8:95) which can't be used:

```
Found pkg-config 'C:\\Strawberry\\perl\\bin\\pkg-config.BAT' but it is Strawberry Perl and thus broken. Ignoring...
Found pkg-config: NO
```

and so it tries to build the subproject dependencies but fails to find Python.h ultimately, because it needs something like [this line](https://github.com/bleachbit/pygtkwin/blob/main/.github/workflows/build-pygobject.yaml#L77).
